Python/bytecodes.c

```diff
@@ -2114,11 +2115,34 @@ dummy_func(
         }
 
         inst(JUMP_BACKWARD, (--)) {
-            assert(oparg < INSTR_OFFSET());
-            JUMPBY(-oparg);
+            _Py_CODEUNIT *here = next_instr-1;
+            assert(oparg <= INSTR_OFFSET());
+            JUMPBY(1-oparg);
```

Comment:
Not too much surgery, but too much for this PR.

We were already restricting the jump range by jumping from `next_instr`, not the instruction, so it might have been worth improving anyway.